numba/tests/test_firstlinefinder.py

```diff
@@ -79,6 +80,32 @@ def test_string_eval(self):
         # Cannot determine first line of string evaled functions
         self.assertIsNone(first_def_line)
 
+    def test_string_eval_linecache(self):
+        source = """def foo():
+            pass
+        """
+
+        # Modify the line cache in a similar manner to Jupyter, so that
+        # get_func_body_first_lineno can find the code using the fallback to
+        # inspect.getsourcelines()
+        filename = "<foo>"
+        timestamp = None
+        entry = (len(source), timestamp, source.splitlines(), filename)
```

Comment:
```suggestion
        entry = (len(source), timestamp, source.splitlines(True), filename)
```
keepends for linecache